src/common/search/SearchPlays.jsx

```diff
@@ -1,36 +1,48 @@
-
-import { useContext } from 'react';
+import { useContext, useRef } from "react";
 import { useLocation, useNavigate } from "react-router-dom";
-import { SearchContext } from './search-context';
-import './search.css';
+import { SearchContext } from "./search-context";
+import "./search.css";
 import { BiSearch } from "react-icons/bi";
+import useBackListener from "common/routing/hooks/useBackListener";
 
 const SearchPlays = () => {
   const location = useLocation();
   const navigate = useNavigate();
-  const {setSearchTerm} = useContext(SearchContext);
-  
+  const { setSearchTerm } = useContext(SearchContext);
+  const inputRef = useRef(null);
+  useBackListener(({ action }) => {
+    if (action === "POP") {
+      inputRef.current.value = "";
```

Comment:
Do we need to set value both by ref and state? Why?

Reply:
input tag value is handled through ref as it's not handled by any state